Vision2022/main.py

```diff
@@ -170,23 +175,31 @@ def send(data):
     s.sendto(data.encode('utf-8'), address)
 
 
-def send_data(hoop_found, hoop_x, hoop_y, hoop_angle, ball_found, ball_x, ball_y, ball_angle, wait_for_other_robot):
+def send_data(hoop_x, hoop_y, hoop_angle, ball_found, ball_x, ball_y, ball_angle, wait_for_other_robot):
+
     data = '%3.1f %3.1f %3.1f %3.1f %3.1f %3.1f %d %d %d \n' % (
-    hoop_x, hoop_y, hoop_angle, ball_x, ball_y, ball_angle, int(ball_found), int(ball_found),
-    wait_for_other_robot)
+        hoop_x, hoop_y, hoop_angle, ball_x, ball_y, ball_angle, int(ball_found), int(ball_found),
+        wait_for_other_robot)
     send(data)
 
 
+def receive():
+    global color
+    global sock
+    data, addr = sock.recvfrom(1024)  # buffer size is 1024 bytes
```

Comment:
Need to handle the case where no data is returned (I think it returns None; a test with no sender will demonstrate). Also this returns bytes and not a string, so have to call decode() on it.